### PR TITLE
Merge "galley" tests into pilot suite

### PR DESCRIPTION
**Pull request by howardjohn**
These tests complete in ~15, but the full overhead of spinning up a test
environment takes quite some time. Its more efficient to just merge
these in, especially since "galley" does not exist anymore

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/pilot/analyze_test.go [hunk 1]
```diff
@@ -387,9 +398,24 @@ func istioctlWithStderr(t *testing.T, i istioctl.Instance, ns string, useKube bo
...
-	if err := cluster.ApplyYAMLFiles(ns, filename); err != nil {
-		t.Fatal(err)
+func applyFileOrFail(ctx framework.TestContext, ns, filename string) interface {
+	Delete()
+} {
+	ctx.Helper()
+	if err := ctx.Clusters().Default().ApplyYAMLFiles(ns, filename); err != nil {
+		ctx.Fatal(err)
 	}
+	return &cleanup{
```
Comment (shamsher31) [inline tests/integration/pilot/analyze_test.go:408]: Is it possible to use it here?
https://github.com/istio/istio/blob/a9a5d5ce347baa6fbc19ccaeebc9daa92f2fec19/pkg/test/failer.go#L30

Comment (howardjohn) [reply tests/integration/pilot/analyze_test.go:409]: Copying a slack comment here for longevity: 

> what do you think the behavior of --nocleanup in integration test should be for config applied during a test? Currently its inconsistent due to different usage patterns (defer vs WhenDone vs Cleanup).
> We can respect the flag, but that means that you cannot run multiple tests (conflicting resources will be created and still remain). Or we can ignore it and always delete yaml, but it makes the flag a bit confusing. Or we can add a third flag. Any thoughts?

Basically we can probably do something better holistically here

**Pull request merged**